code/drasil-lang/lib/Language/Drasil/Document/Core.hs

```diff
@@ -69,13 +69,16 @@ data RawContent =
   | EqnBlock ModelExpr                       -- ^ Block of Equations holds an expression.
   | DerivBlock Sentence [RawContent]         -- ^ Grants the ability to label a group of 'RawContent'.
   | Enumeration ListType                     -- ^ For enumerated lists.
-  | Defini DType [(Sentence, [Contents])]    -- ^ Defines something with a type, identifier, and 'Contents'.
+  | Defini DType [(Identifier, [Contents])]    -- ^ Defines something with a type, identifier, and 'Contents'.
   | Figure Lbl Filepath MaxWidthPercent      -- ^ For creating figures in a document. Should use relative file path.
   | Bib BibRef                               -- ^ Grants the ability to reference something.
   | Graph [(Sentence, Sentence)] (Maybe Width) (Maybe Height) Lbl -- ^ Contain a graph with coordinates ('Sentence's), maybe a width and height, and a label ('Sentence').
   | CodeBlock CodeExpr                       -- ^ Block for codes
                -- TODO: Fill this one in.
 
+-- |  An identifier is just a 'Sentence'.
+type Identifier = Sentence
```

Comment:
This is not what I meant. You should be able to use complete `Sentence`s as `Indentifier`s.  Only a fairly restricted set of constructors should be allowed.

The only reasonable way to implement that is as a new type (via `data`) that has functions to transform it to a `Sentence` in latter passes.

The way forward will need to first figure out who creates `Defini` instances, and why are they (now) using a `Sentence` instead of something simpler.